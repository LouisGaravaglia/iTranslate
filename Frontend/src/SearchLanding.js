import React, {useState, useRef, useEffect, useCallback} from 'react';
import {useSpring, animated, useTransition} from 'react-spring';
import {Spring} from 'react-spring/renderprops';
//COMPONENT IMPORTS
import SearchBar from "./SearchBar";
import FlashMessage from "./FlashMessage";
import LanguageSelect from "./LanguageSelect";
import LyricsTranslation from "./LyricsTranslation";
import Tracks from "./Tracks";
//REDUX IMPORTS
import {useDispatch, useSelector} from "react-redux";
import {resetLanguageError, resetTranslationError, resetLyricsError, resetSearchError} from "./actionCreators/handleErrorsCreator";
import {setResultsArray} from "./actionCreators/setResultsArrayCreator";
import {resetStore} from "./actionCreators/resetStoreCreator";
import useOnScreen from "./useOnScreen";

const SearchLanding = ({handleTrackSearchSubmit}) => {
  //STATE FOR ANIMATIONS
  const [bgColor, setBgColor] = useState("#ABA800");
  //REDUX STORE

 
////////////////////////////////////////////////////  RETURN  ////////////////////////////////////////////////////

  return (
    <Spring
      from={{opacity: 1, marginLeft: -2500}}
      to={{opacity: 1, marginLeft: 0}}
      reverse={false}
      config={{delay: 0, duration: 200}}
    >
    {props => (
      <div style={props}>
  

        <div>
      <SearchBar header="Find your song!" handleSubmit={handleTrackSearchSubmit}/>

   {/* <div className="Browse-Landing">
            <button >Artists</button>
            <button >Genre</button>
            <button >Danceability</button>
          </div> */}
    

</div>

      </div>
    )}
    </Spring>
  );
}

export default SearchLanding;