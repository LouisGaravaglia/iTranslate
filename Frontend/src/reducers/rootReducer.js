import {combineReducers} from "redux";
import translationReducer from "./translationReducer";
import errorsReducer from "./errorsReducer";
import lyricsReducer from "./lyricsReducer";
import resultsReducer from "./resultsReducer";

const rootReducer = combineReducers({results: resultsReducer, translation: translationReducer, lyrics: lyricsReducer, errors: errorsReducer});

export default rootReducer;