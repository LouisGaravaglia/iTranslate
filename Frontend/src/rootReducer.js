// const SEED_ARTISTS = [{name: "J Balvin", id:"1vyhD5VmyZ7KMfW5gqLgo5"}, {name: "Bad Bunny", id:"4q3ewBCX7sLwd24euuV69X"}, {name: "Maluma", id:"1r4hJ1h58CWwUQe3MxPuau"}, {name: "Rosal√≠a", id:"7ltDVBr6mKbRvohxheJ9h1"}];
// const INITIAL_STATE = {songs:[], artists: SEED_ARTISTS, albums:[]};
import {GET_TRANSLATION} from "./actionTypes";
const ERRORS = {searchError: false, lyricsError: false, translationError: false}
const INITIAL_STATE = {translation: "", errors: ERRORS }

function rootReducer(state=INITIAL_STATE, action) {
    switch(action.type) {
        case GET_TRANSLATION:
        
            // const updatedStateWithErrors = {...state, state.errors.lyricsError = action.errors.lyricsError}
        return {...state, translation: action.translation}

        default:
            return state;
    }
}

export default rootReducer;
