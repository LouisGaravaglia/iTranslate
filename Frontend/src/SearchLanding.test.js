import React from 'react'
import {render, fireEvent, waitFor, screen, cleanup} from '@testing-library/react'
import {MemoryRouter} from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect'
import { Provider } from 'react-redux'
import { createStore } from "redux";
import rootReducer from "./reducers/rootReducer";
import SearchLanding from "./SearchLanding";



afterEach(cleanup);

const startingState = {};

//REDUCER FUNCTION TO MIMIC REDUX REDUCER FOR STORE
function reducer(state = startingState, action) {
  switch (action.type) {
    case "ADD_TRACKS":
      return {...state, tracks: action.tracks};
    default:
      return state;
  };
};

//FUNCTION TO ALLOW THE COMPONENT TO BE RENDERED USING OUR MAKESHIF REDUX STORE
function renderWithRedux(
  component,
  {initialState, store = createStore(reducer, initialState)} = {}
) {
  return {
    ...render(<Provider store={store}>{component}</Provider>)
  };
};

//SMOKE TEST
describe('Smoke Test for SearchLanding component', () => {

  it('renders without crashing', () => {
    renderWithRedux(<SearchLanding />)
  });

});


