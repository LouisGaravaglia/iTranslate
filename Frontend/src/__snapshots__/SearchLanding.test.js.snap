// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test for SearchLanding component matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="opacity: 1; margin-left: -2500px;"
  >
    <div
      class="Search-Field"
    >
      <div
        class="Search-Field-Content"
      >
        <h1>
          Find your song!
        </h1>
        <form
          class="Search-Input-Container"
        >
          <div>
            <input
              id="SearchVal"
              name="searchVal"
              type="text"
              value=""
            />
            <button
              type="submit"
            >
              <i
                class="fa fa-search icon"
              />
            </button>
          </div>
        </form>
      </div>
      <div
        class="Search-Field-Filler"
      />
    </div>
  </div>
</DocumentFragment>
`;
